synapse/handlers/message.py

```diff
@@ -901,6 +912,31 @@ def persist_and_notify_client_event(
                         "Room alias %s does not point to the room" % (room_alias_str,),
                     )
 
+            # Check that alt_aliases is the proper form.
+            alt_aliases = event.content.get("alt_aliases", [])
+            if not isinstance(alt_aliases, (list, tuple)):
+                raise SynapseError(400, "The alt_aliases property must be a list.")
+
+            # If the old version of alt_aliases is of an unknown form,
+            # completely replace it.
+            if not isinstance(original_alt_aliases, (list, tuple)):
+                original_alt_aliases = []
+
+            # Check that each alias is currently valid.
+            new_alt_aliases = set(alt_aliases) - set(original_alt_aliases)
+            if new_alt_aliases:
+                for alias_str in new_alt_aliases:
+                    room_alias = RoomAlias.from_string(alias_str)
+                    directory_handler = self.hs.get_handlers().directory_handler
```

Comment:
can we avoid doing this in each cycle of the loop?

generally we try to do this sort of thing in the constructor, to pick up errors at startup, however it may be here to work around import cycles. But at least pull it out a bit.

Reply:
We should share it with the checking of the alias property above too. 👍

Reply:
I attempted to move this to the `__init__` and it caused an import cycle as you expected. I modified the code to not get the handler each time though.